src/solver/utils/ortools_utils.cpp

```diff
@@ -125,12 +125,8 @@ MPSolver* convert_to_MPSolver(
 {
     auto& study = *Data::Study::Current::Get();
 
-    // Define solver used depending on study option
-    MPSolver::OptimizationProblemType solverType
-      = OrtoolsUtils().getLinearOptimProblemType(study.parameters.ortoolsEnumUsed);
-
-    // Create the linear solver instance
-    MPSolver* solver = new MPSolver("simple_lp_program", solverType);
+    // Create the MPSolver
+    MPSolver* solver = factory(problemeSimplexe, study.parameters.ortoolsSolver);
```

Comment:
1. Using a factory here is a good idea
2. `factory` is too generic, choose a more specific name (it can contain factory)